calculator-data/src/test/java/camp/nextstep/edu/calculator/data/CalculatorReadWriteTest.kt

```diff
@@ -0,0 +1,46 @@
+package camp.nextstep.edu.calculator.data
+
+import android.content.Context
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import java.io.IOException
+import kotlin.concurrent.thread
+
+@RunWith(RobolectricTestRunner::class)
+class CalculatorReadWriteTest {
+    private lateinit var historyDao: HistoryDao
+    private lateinit var db: HistoryDatabase
+
+    @Before
+    fun createDb() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        db = Room.inMemoryDatabaseBuilder(
+            context,
+            HistoryDatabase::class.java
+        ).build()
+        historyDao = db.historyDao()
+    }
```

Comment:
테스트에서는 mainThread에서 디비가 동작할 수 있게 만들어주는 건 어떨까요?
```suggestion
    fun createDb() {
        val context = ApplicationProvider.getApplicationContext<Context>()
        db = Room.inMemoryDatabaseBuilder(
            context,
            HistoryDatabase::class.java
        )
        .allowMainThreadQueries()
        .build()
        historyDao = db.historyDao()
    }
```

Reply:
DB를 mainThread에서 생성하면서 얻을 수 있는 이점이 어떤게 있을까요?

Reply:
테스트에서 DAO를 접근할 때 매번 스레들을 옮겨주어야하기에, 이런 반복적인 작업을 줄여볼 수 있겠지요 :)
지금은 suspend로 변경되었기에, Room이 자동으로 옮겨주어 `.allowMainThreadQueries()`를 추가할 필요가 없습니다 :)